generators/utilities/src/main/resources/csrc/emulator.cc

```diff
@@ -51,9 +57,14 @@ double sc_time_stamp()
   return trace_count;
 }
 
-extern "C" int vpi_get_vlog_info(void* arg)
+// enable verilator multithreading
+static int htif_argc;
+static char **htif_argv = NULL;
+extern "C" int vpi_get_vlog_info(s_vpi_vlog_info *vlog_info_s)
```

Comment:
I'm sure there's a reason for passing the HTIF args this way, but would you mind explaining it? Are we passing the multithreading args through VPI somehow?

Reply:
we want a way for the thread that calls `SimSerial's tick()` from within verilog to have access to the cmdline args to initialize the HTIF object. due to multithreading

Reply:
I actually can't find where the `htif` object is created now (I see where it used to be). Am I missing something obvious?

Reply:
SimSerial.cc and friends creates it on the 1st tick

Reply:
I see where `tsi` is created (SimSerial.cc) but not `htif` or `jtag`

Reply:
htif is an interface that tsi implements.
look at the `SimSerial.cc`, `SimJTAG.cc` and `SimDTM.cc` files

Reply:
Sorry, I meant `dtm` in both of those comments, but yea, they're in rocket-chip, not TCIP, so I found them now. Looks good to me.